### DISCUSSION TITLE
Feature request: Scan containers for vulnerabilities

### DISCUSSION THREAD

**Discussion by kript** [wtsi-hgi/softpack-docs#9] [Design Review]
There are a number of tools available to this project which could be used to do automated CI based Static Code Analysis against the SoftPack Builder artefacts such as Singularity (or Docker) Containers. Example are;

* Dependabot
* Snyk.io using the existing Sanger tooling

Having the CI check the containers as they are created and then over time would help flag vulnerable packages or security issues within the container themselves, and allow the HumGen Informatics team to either update the containers to let the researcher know of the vulnerability and any implications for their workflow or results..

**Comment by altaf-ali** [comment]
Yes, this is a good idea. I'll add it to the list of features to be implemented.
